src/main/geometries/Sphere.java

```diff
@@ -47,7 +51,25 @@ public NormalizedVector normal(Point p) {
 
 	@Override
 	public List<Point> intersect(Ray ray) {
-		// TODO Auto-generated method stub
-		return null;
+		Vector centerToRaySource;
```

Comment:
This can be done in ctor

Reply:
What, declaring the variable? That takes literally zero effort by the compiler. When the function gets called, the stack pointer is moved up to make room for all the local variables. It doesn't matter how many there are, it's just addition. The number to add is calculated at compile time. If you meant something else, lmk

Reply:
Ah sorry I thought it was going to origin. U right